USAGE.md

```diff
@@ -188,6 +188,41 @@ implementing the DataStore API and can be used interchangeably with the recipes
 
 This section utilizes the [stix2 python library](https://github.com/oasis-open/cti-python-stix2). Please refer to the [STIX2 Python API Documentation](https://stix2.readthedocs.io/en/latest/) for more information on how to work with STIX programmatically.
 
+## Requirements and imports
+Before installing requirements, we recommend setting up a virtual environment:
+1. Create virtual environment:
+    - macOS and Linux: `python3 -m venv env`
+    - Windows: `py -m venv env`
+2. Activate the virtual environment:
+    - macOS and Linux: `source env/bin/activate`
+    - Windows: `env/Scripts/activate.bat`
+
+### stix2
+[stix2 can be installed by following the instructions on their repository](https://github.com/oasis-open/cti-python-stix2#installation). Imports for the recipes in this repository can be done from the base package, for example:
+
+```python
+from stix2 import Filter
+```
+
+However, if you are aiming to extend the ATT&CK dataset with new objects or implement complex workflows, you may need to use the `v20` specifier for some imports. This ensures that the objects use the STIX 2.0 API instead of the STIX 2.1 API. For example:
+
+```python
+from stix2.v20 import AttackPattern
+```
+
+You can see a full list of the classes which have versioned imports [here](https://stix2.readthedocs.io/en/latest/api/stix2.v20.html).
+
+### taxii2client
+[taxii2-client can be installed by following the instructions on their repository](https://github.com/oasis-open/cti-taxii-client#installation). How taxii2client should be imported depends on what version you have installed. You can check your installed version using `pip3 show taxii2-client | grep Version`.
+- **If version >= 2.0.0**, you *must* use the `v20` import in order to avoid [HTTP 406](https://httpstatuses.com/406) responses when connecting to the ATT&CK TAXII server:
+    ```python
+    from taxii2client.v20 import Collection
+    ```
+- **If version < 2.0.0**, you *must not* use the `v20` import because it was introduced in version 2.0.0:
+    ```python
+    from taxii2client import Collection
+    ```
+
```

Comment:
There are two key points in this section of the USAGE relevant to the users.
- Why do we have to import `v20` version of the client? MITRE TAXII Server currently implements the 2.0 version of the TAXII specification. The usage document speaks to the fact there is a Server but no version details are given.
- The import problem you added as new text. Both `stix2` and `taxii2client` were built with the design philosophy provide multi specification support via one distributable per spec. Ideally we would encourage people to always be on the latest version because the distributables are meant support STIX 2.0/2.1/2.X+ for `stix2` or TAXII 2.0/2.1/2.X+ for `taxii2client`.

The text could be changed to perhaps convey ```If the TAXII Client is getting a 406 Response, make sure you are running the latest version. In addition, make sure you are running the 2.0 version of the client in order to communicate with the ATT&CK TAXII Server.``` This would help in removing the `If version < 2.0.0` part of the text.

On our end we will also add a version restriction to this dependency.

Reply:
All good points, which I have addressed in the my latest revision of this section.